<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>bulk-RNA seq测序数据分析流程 - IT学习者博客</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
		<meta property="og:title" content="bulk-RNA seq测序数据分析流程" />
<meta property="og:description" content="假如有bulk-RNA测序的数据：TH1，TH2，TH3三个重复（实验组），TW1，TW2，TW3三个重复（对照组）
准备工作 需要安装的软件（如FastQC、Trimmomatic、HISAT2、StringTie、samtools） conda install -c bioconda fastqc conda install trimmomatic conda install -c bioconda stringtie conda install -c bioconda hisat2 conda install samtools conda install -c bioconda picard 1. 质量控制 首先，需要对原始的测序数据（FASTQ格式）进行质量控制。通常使用FastQC进行初步的质量检查，然后使用Trimmomatic或者其他工具进行质量修剪。
# 质量检查 fastqc *.fq.gz # 质量修剪 for sample in TH1 TH2 TH3 TW1 TW2 TW3 do trimmomatic PE ${sample}.R1.fq.gz ${sample}.R2.fq.gz \ ${sample}.R1.trim.fq.gz ${sample}.R1.untrim.fq.gz \ ${sample}.R2.trim.fq.gz ${sample}.R2.untrim.fq.gz \ SLIDINGWINDOW:4:20 MINLEN:25 done 2.读段比对 使用比对工具（如HISAT2, STAR等）将质量修剪后的读段比对到参考基因组。
使用HISAT2进行读段比对 # 假设参考基因组索引为genome_index for sample in TH1 TH2 TH3 TW1 TW2 TW3 do hisat2 -x genome_index -1 ${sample}." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://itnewbiea.github.io/posts/ae1fc04465ea589a98be6d6cdc55ff63/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2024-01-06T15:10:33+08:00" />
<meta property="article:modified_time" content="2024-01-06T15:10:33+08:00" />


	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">
		
</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo">
		<a class="logo__link" href="/" title="IT学习者博客" rel="home">
			<div class="logo__item logo__text">
					<div class="logo__title">IT学习者博客</div>
					
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">bulk-RNA seq测序数据分析流程</h1>
			
		</header>
		<div class="content post__content clearfix">
			
<div id="content_views" class="markdown_views prism-atom-one-dark">
                    <svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
                        <path stroke-linecap="round" d="M5,0 0,2.5 5,5z" id="raphael-marker-block" style="-webkit-tap-highlight-color: rgba(0, 0, 0, 0);"></path>
                    </svg>
                    <p>假如有bulk-RNA测序的数据：TH1，TH2，TH3三个重复（实验组），TW1，TW2，TW3三个重复（对照组）</p> 
<ul><li>准备工作</li></ul> 
<pre><code>需要安装的软件（如FastQC、Trimmomatic、HISAT2、StringTie、samtools）
conda install -c bioconda fastqc
conda install trimmomatic
conda install -c bioconda stringtie 
conda install -c bioconda hisat2
conda install samtools
conda install -c bioconda picard
</code></pre> 
<h2><a id="1__14"></a>1. 质量控制</h2> 
<p>首先，需要对原始的测序数据（FASTQ格式）进行质量控制。通常使用FastQC进行初步的质量检查，然后使用Trimmomatic或者其他工具进行质量修剪。</p> 
<pre><code class="prism language-bash"><span class="token comment"># 质量检查</span>
fastqc *.fq.gz

<span class="token comment"># 质量修剪</span>
<span class="token keyword">for</span> <span class="token for-or-select variable">sample</span> <span class="token keyword">in</span> TH1 TH2 TH3 TW1 TW2 TW3
<span class="token keyword">do</span>
    trimmomatic PE <span class="token variable">${sample}</span>.R1.fq.gz <span class="token variable">${sample}</span>.R2.fq.gz <span class="token punctuation">\</span>
    <span class="token variable">${sample}</span>.R1.trim.fq.gz <span class="token variable">${sample}</span>.R1.untrim.fq.gz <span class="token punctuation">\</span>
    <span class="token variable">${sample}</span>.R2.trim.fq.gz <span class="token variable">${sample}</span>.R2.untrim.fq.gz <span class="token punctuation">\</span>
    SLIDINGWINDOW:4:20 MINLEN:25
<span class="token keyword">done</span>
</code></pre> 
<h2><a id="2_31"></a>2.读段比对</h2> 
<p>使用比对工具（如HISAT2, STAR等）将质量修剪后的读段比对到参考基因组。</p> 
<ul><li>使用HISAT2进行读段比对</li></ul> 
<pre><code class="prism language-bash"><span class="token comment"># 假设参考基因组索引为genome_index</span>
<span class="token keyword">for</span> <span class="token for-or-select variable">sample</span> <span class="token keyword">in</span> TH1 TH2 TH3 TW1 TW2 TW3
<span class="token keyword">do</span>
    hisat2 <span class="token parameter variable">-x</span> genome_index <span class="token parameter variable">-1</span> <span class="token variable">${sample}</span>.R1.trim.fq.gz <span class="token parameter variable">-2</span> <span class="token variable">${sample}</span>.R2.trim.fq.gz <span class="token punctuation">\</span>
    <span class="token parameter variable">-S</span> <span class="token variable">${sample}</span>.sam
<span class="token keyword">done</span>

</code></pre> 
<ul><li>使用STAR进行读段比对</li></ul> 
<ol><li>下载并构建STAR索引</li></ol> 
<p>在使用STAR进行比对之前，需要首先构建或下载参考基因组的索引。这是一个一次性的过程。假设你已经有了参考基因组的fasta文件和基因注释文件（GTF格式），使用以下命令构建索引：</p> 
<pre><code class="prism language-bash">STAR <span class="token parameter variable">--runThreadN</span> NumberOfThreads <span class="token punctuation">\</span>
<span class="token parameter variable">--runMode</span> genomeGenerate <span class="token punctuation">\</span>
<span class="token parameter variable">--genomeDir</span> /path/to/genomeDir <span class="token punctuation">\</span>
<span class="token parameter variable">--genomeFastaFiles</span> /path/to/genome/fasta <span class="token punctuation">\</span>
<span class="token parameter variable">--sjdbGTFfile</span> /path/to/annotations.gtf <span class="token punctuation">\</span>
<span class="token parameter variable">--sjdbOverhang</span> ReadLength-1
</code></pre> 
<p>其中，NumberOfThreads是使用的线程数，/path/to/genomeDir是存储索引的目录，/path/to/genome/fasta和/path/to/annotations.gtf分别是参考基因组的fasta文件和基因注释文件的路径。ReadLength-1是读段长度减去1，这是一个重要的参数，需要根据数据来设定。<br> 2. 读段比对</p> 
<pre><code class="prism language-bash"><span class="token keyword">for</span> <span class="token for-or-select variable">sample</span> <span class="token keyword">in</span> TH1 TH2 TH3 TW1 TW2 TW3
<span class="token keyword">do</span>
    STAR <span class="token parameter variable">--genomeDir</span> /path/to/genomeDir <span class="token punctuation">\</span>
    <span class="token parameter variable">--readFilesIn</span> <span class="token variable">${sample}</span>.R1.trim.fq.gz <span class="token variable">${sample}</span>.R2.trim.fq.gz <span class="token punctuation">\</span>
    <span class="token parameter variable">--runThreadN</span> <span class="token number">10</span> <span class="token punctuation">\</span>
    <span class="token parameter variable">--outFileNamePrefix</span> <span class="token variable">${sample}</span> <span class="token punctuation">\</span>
    <span class="token parameter variable">--readFilesCommand</span> zcat <span class="token punctuation">\</span>
    <span class="token parameter variable">--outSAMtype</span> BAM SortedByCoordinate
<span class="token keyword">done</span>
</code></pre> 
<p>在这里，/path/to/genomeDir是你的索引目录，NumberOfThreads是使用的线程数，<span class="katex--inline"><span class="katex"><span class="katex-mathml"> 
     
      
       
        
        
          s 
         
        
          a 
         
        
          m 
         
        
          p 
         
        
          l 
         
        
          e 
         
        
       
         . 
        
       
         R 
        
       
         1. 
        
       
         t 
        
       
         r 
        
       
         i 
        
       
         m 
        
       
         . 
        
       
         f 
        
       
         q 
        
       
         . 
        
       
         g 
        
       
         z 
        
       
         和 
        
       
      
        {sample}.R1.trim.fq.gz和 
       
      
    </span><span class="katex-html"><span class="base"><span class="strut" style="height: 0.8889em; vertical-align: -0.1944em;"></span><span class="mord"><span class="mord mathnormal">s</span><span class="mord mathnormal">am</span><span class="mord mathnormal" style="margin-right: 0.0197em;">pl</span><span class="mord mathnormal">e</span></span><span class="mord">.</span><span class="mord mathnormal" style="margin-right: 0.0077em;">R</span><span class="mord">1.</span><span class="mord mathnormal">t</span><span class="mord mathnormal" style="margin-right: 0.0278em;">r</span><span class="mord mathnormal">im</span><span class="mord">.</span><span class="mord mathnormal" style="margin-right: 0.1076em;">f</span><span class="mord mathnormal" style="margin-right: 0.0359em;">q</span><span class="mord">.</span><span class="mord mathnormal" style="margin-right: 0.0359em;">g</span><span class="mord mathnormal" style="margin-right: 0.044em;">z</span><span class="mord cjk_fallback">和</span></span></span></span></span>{sample}.R2.trim.fq.gz分别是质量修剪后的读段文件。–outSAMtype BAM SortedByCoordinate会输出排序后的BAM文件，这对后续分析非常有用。</p> 
<ul><li> <p>HISAT2和STAR的区别如下：</p> 
  <ol><li> <p>设计和算法：<br> HISAT2：使用了一种称为分层索引的方法。它首先将参考基因组划分为较小的片段，并为这些片段构建全局和局部的索引。这种方法使得HISAT2在处理基因组比对时特别有效，尤其是在考虑剪接位点和变异位点时。<br> STAR：使用了一种称为基于稀疏后缀数组的方法。STAR的这种方法允许它在一个单独的步骤中完成比对和剪接位点的检测，这使得它在处理大型数据集时非常快速和内存高效。</p> </li><li> <p>性能和资源消耗：<br> HISAT2：通常比STAR使用更少的内存，但在处理时间上可能稍长。它特别适合于资源受限的计算环境。<br> STAR：虽然需要较多的内存，但在执行速度上通常比HISAT2更快，特别是在处理包含大量剪接事件的复杂转录组时。</p> </li><li> <p>准确性和灵敏度：<br> HISAT2：在处理较长的读段和考虑基因组变异（如SNPs和小的indels）时表现出较高的准确性。<br> STAR：在检测剪接事件和转录本多样性方面表现优异，尤其是在处理短读段数据时。</p> </li><li> <p>适用性： HISAT2：由于其对于基因组变异的考虑，HISAT2在进行种群遗传学研究或者需要高精度比对的场景中表现更好。 STAR：在进行转录组学研究，尤其是需要快速处理大规模数据集和复杂剪接事件的情况下更为适用。</p> </li><li> <p>易用性和功能： 两者都具备易用性和丰富的功能集，包括对多线程的支持、与常用转录组分析流程的兼容性等。</p> </li></ol> </li></ul> 
<p>综上，选择HISAT2还是STAR取决于你的具体需求，比如数据的类型、可用资源、分析的焦点等。在某些情况下，研究者可能会根据不同阶段的需求选择不同的工具。例如，在初步筛选或快速分析时使用STAR，而在需要更细致分析基因组变异或长读段数据时使用HISAT2。</p> 
<h2><a id="3PCR_93"></a>3.标记和去除PCR重复</h2> 
<p>使用Picard或samtools来标记和去除重复的读段。<br> 对于使用samtools，流程如下：</p> 
<pre><code class="prism language-bash"><span class="token keyword">for</span> <span class="token for-or-select variable">sample</span> <span class="token keyword">in</span> TH1 TH2 TH3 TW1 TW2 TW3
<span class="token keyword">do</span>
    samtools view <span class="token parameter variable">-bS</span> <span class="token variable">${sample}</span>.sam <span class="token operator">|</span> <span class="token punctuation">\</span>
    samtools <span class="token function">sort</span> <span class="token parameter variable">-o</span> <span class="token variable">${sample}</span>.sorted.bam
    
    samtools markdup <span class="token variable">${sample}</span>.sorted.bam <span class="token variable">${sample}</span>.dedup.bam
<span class="token keyword">done</span>
</code></pre> 
<p>对于使用Picard, 流程如下：</p> 
<pre><code class="prism language-bash"><span class="token keyword">for</span> <span class="token for-or-select variable">sample</span> <span class="token keyword">in</span> TH1 TH2 TH3 TW1 TW2 TW3
<span class="token keyword">do</span>
    samtools view <span class="token parameter variable">-bS</span> <span class="token variable">${sample}</span>.sam <span class="token operator">|</span> <span class="token punctuation">\</span>
    samtools <span class="token function">sort</span> <span class="token parameter variable">-o</span> <span class="token variable">${sample}</span>.sorted.bam

    picard MarkDuplicates <span class="token assign-left variable">I</span><span class="token operator">=</span><span class="token variable">${sample}</span>.sorted.bam <span class="token punctuation">\</span>
    <span class="token assign-left variable">O</span><span class="token operator">=</span><span class="token variable">${sample}</span>.dedup.bam <span class="token assign-left variable">M</span><span class="token operator">=</span><span class="token variable">${sample}</span>.marked_dup_metrics.txt
<span class="token keyword">done</span>
</code></pre> 
<h2><a id="4_120"></a>4.转录组组装和表达量估计</h2> 
<p>使用工具如StringTie或Cufflinks对SAM/BAM文件进行转录组组装，生成转录本。</p> 
<pre><code class="prism language-bash">
<span class="token keyword">for</span> <span class="token for-or-select variable">sample</span> <span class="token keyword">in</span> TH1 TH2 TH3 TW1 TW2 TW3
<span class="token keyword">do</span>
    stringtie <span class="token parameter variable">-e</span> <span class="token parameter variable">-B</span> <span class="token parameter variable">-G</span> reference.gtf <span class="token parameter variable">-o</span> <span class="token variable">${sample}</span>.gtf <span class="token variable">${sample}</span>.dedup.bam
<span class="token keyword">done</span>

</code></pre> 
<p>-e 选项告诉StringTie只估计已知转录本（在提供的GTF文件中）的表达量。<br> -B 选项会生成每个基因和转录本的表达量计数文件，这些文件可用于后续差异表达分析。<br> -G reference.gtf 是参考基因组的注释文件，这里需要你提供正确的路径和文件名。</p> 
<h2><a id="5_134"></a>5.差异表达分析</h2> 
<p>最后，使用DESeq2或edgeR等工具进行差异表达分析，以识别实验组和对照组之间差异表达的基因。</p> 
<pre><code class="prism language-bash">library<span class="token punctuation">(</span>DESeq2<span class="token punctuation">)</span>

<span class="token comment"># 假设countData为一个矩阵，行为基因，列为样本，包含基因的计数</span>
<span class="token comment"># colData为DataFrame，包含样本信息</span>
<span class="token comment"># 这里需要根据实际情况准备countData和colData</span>

dds <span class="token operator">&lt;</span>- DESeqDataSetFromMatrix<span class="token punctuation">(</span>countData <span class="token operator">=</span> countData,
                              colData <span class="token operator">=</span> colData,
                              design <span class="token operator">=</span> ~ condition<span class="token punctuation">)</span>

dds <span class="token operator">&lt;</span>- DESeq<span class="token punctuation">(</span>dds<span class="token punctuation">)</span>
res <span class="token operator">&lt;</span>- results<span class="token punctuation">(</span>dds<span class="token punctuation">)</span>

<span class="token comment"># 保存结果</span>
write.csv<span class="token punctuation">(</span>as.data.frame<span class="token punctuation">(</span>res<span class="token punctuation">)</span>, <span class="token assign-left variable">file</span><span class="token operator">=</span><span class="token string">"DESeq2_results.csv"</span><span class="token punctuation">)</span>
</code></pre>
                </div>
		</div>
	</article>
</main>


<nav class="pager flex">
	<div class="pager__item pager__item--prev">
		<a class="pager__link" href="/posts/1a44a5026bbda42fdd82193ba0ccd8c6/" rel="prev">
			<span class="pager__subtitle">«&thinsp;Previous</span>
			<p class="pager__title">自动驾驶感知-预测-决策-规划-控制学习（3）：感知方向文献阅读笔记</p>
		</a>
	</div>
	<div class="pager__item pager__item--next">
		<a class="pager__link" href="/posts/6e876532bd96350a0908a8f631265b53/" rel="next">
			<span class="pager__subtitle">Next&thinsp;»</span>
			<p class="pager__title">C# Unity将地形(Terrain)导出成obj文件</p>
		</a>
	</div>
</nav>


			</div>
			
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2024 IT学习者博客.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>
<script src="https://itnewbiea.github.io/js/foot.js"></script>


	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>